### PR TITLE
Seanmcm/debug fixes

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Still need to more testing (on Mac...)

### CODE REVIEW

File: Extension/src/Debugger/configurationProvider.ts [hunk 1]
```diff
@@ -127,7 +126,7 @@ class CppConfigurationProvider implements vscode.DebugConfigurationProvider {
                 const suffix: string = suffixIndex === -1 ? "" : compilerName.substr(suffixIndex);
                 debuggerName = (platform === "darwin" ? "lldb" : "lldb-mi") + suffix;
             } else {
-                debuggerName = "gdb";
+                debuggerName = platform === "win32" ? "gdb.exe" : "gdb";
```
Comment (bobbrow) [inline Extension/src/Debugger/configurationProvider.ts]: Do we need to do something similar for lldb on Windows?

File: Extension/src/Debugger/extension.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export function initialize(context: vscode.ExtensionContext): void {
         disposables.push(vscode.debug.registerDebugConfigurationProvider('cppvsdbg', new CppVsDbgConfigurationProvider(configurationProvider)));
     }
     const provider: CppDbgConfigurationProvider = new CppDbgConfigurationProvider(configurationProvider);
-    disposables.push(vscode.debug.registerDebugConfigurationProvider('cppdbg', provider)); //new QuickPickConfigurationProvider(provider))); // TODO
+    disposables.push(vscode.debug.registerDebugConfigurationProvider('cppdbg', new QuickPickConfigurationProvider(provider))); // TODO
```
Comment (bobbrow) [inline Extension/src/Debugger/extension.ts]: remove "TODO" comment?

File: Extension/src/Debugger/configurationProvider.ts [hunk 3]
```diff
@@ -93,8 +92,8 @@ class CppConfigurationProvider implements vscode.DebugConfigurationProvider {
 	 * Returns a list of initial debug configurations based on contextual information, e.g. package.json or folder.
 	 */
     async provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration[]> {
-        let buildTasks: vscode.Task[]; // = await getBuildTasks(); // TODO
-        if (!buildTasks || buildTasks.length === 0) {
+        let buildTasks: vscode.Task[] = await getBuildTasks(); // TODO
```
Comment (bobbrow) [inline Extension/src/Debugger/configurationProvider.ts]: You should try/catch this.  If the promise is rejected, you should get an exception here, not an empty array.

File: Extension/src/Debugger/configurationProvider.ts [hunk 4]
```diff
@@ -57,9 +56,9 @@ export class QuickPickConfigurationProvider implements vscode.DebugConfiguration
 
         return vscode.window.showQuickPick(items, {placeHolder: "Select a configuration"}).then(async selection => {
             // Wrap in new Promise to make sure task kicks off before VS Code switches the active document to launch.json
-            return new Promise<vscode.DebugConfiguration[]>(async resolve => {
+            return new Promise<vscode.DebugConfiguration[]>(async (resolve, reject) => {
                 if (!selection) {
-                    return resolve([defaultConfig]); // User canceled it. Choose the default config?
+                    return reject([]); // User canceled it.
```
Comment (bobbrow) [inline Extension/src/Debugger/configurationProvider.ts]: If you want to return an empty array, the way to do it is with `resolve([])`, not `reject([])`.  reject goes through the error callback (or throws an exception if you're using `await`)

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/src/Debugger/configurationProvider.ts (hunk 4 above)
Comment (sean-mcmanus) [reply Extension/src/Debugger/configurationProvider.ts]: Griffin was saying that was intentional. I'll discuss it with you tomorrow.

File: Extension/src/Debugger/configurationProvider.ts (hunk 3 above)
Comment (sean-mcmanus) [reply Extension/src/Debugger/configurationProvider.ts]: The promise is never rejected, right?

**Review by bobbrow — APPROVED** [review]

File: Extension/src/Debugger/configurationProvider.ts (hunk 4 above)
Comment (bobbrow) [reply Extension/src/Debugger/configurationProvider.ts]: There is an async anti-pattern in here.  We talked about it and replaced the promises with async/await

**Pull request merged**